cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+	"sync"
+
+	gce "google.golang.org/api/compute/v1"
+	"k8s.io/client-go/util/workqueue"
+	klog "k8s.io/klog/v2"
+)
+
+// MigInfoProvider allows obtaining information about MIGs
+type MigInfoProvider interface {
```

Comment:
This refactor could be an opportunity to add some unittests here. I think it should be fairly easy given that the only external dependency seems to be AutoscalingGceClient, which is an interface and so could be easily mocked. 

I'm not going to block this PR given that we didn't have tests for previous implementation, but I think tests would be a really nice addition if you're up for it.

Reply:
That's a good point, I'll add a commit with tests to this PR. I'm already immersed in this code so postponing it for later seems like a waste of context.

Reply:
Done